### PR TITLE
Variant lowlevel class

### PR DESCRIPTION
**Pull request by benjeffery**
Part of #605

### CODE REVIEW

**Comment by benjeffery** [comment]
@mergifyio rebase

**Comment by benjeffery** [comment]
@jeromekelleher This is the basics of the Variant class -`restricted_copy` coming in the next PR.

**Comment by benjeffery** [comment]
Note that the missing coverage should be fixed when the vargen tests are bought across.

File: python/_tskitmodule.c [hunk 1]
```diff
@@ -11295,6 +11301,221 @@ static PyTypeObject VariantGeneratorType = {
...
+    static char *kwlist[] = { "site", NULL };
+
+    if (Variant_check_state(self) != 0) {
+        goto out;
+    }
+    if (!PyArg_ParseTupleAndKeywords(
+            args, kwds, "O&", kwlist, &tsk_id_converter, &site_id)) {
+        goto out;
+    }
+    err = tsk_variant_decode(self->variant, site_id, 0);
```
Comment (jeromekelleher) [inline python/_tskitmodule.c:11416]: Should get test coverage on this - won't be covered by vargen as you can't specify the ID.

Easy to do, just pass in a bad site ID.

File: python/_tskitmodule.c [hunk 2]
```diff
@@ -11295,6 +11301,221 @@ static PyTypeObject VariantGeneratorType = {
...
+    array = (PyArrayObject *) PyArray_SimpleNewFromData(
+        1, &dims, NPY_INT32, self->variant->genotypes);
+    if (array == NULL) {
+        goto out;
+    }
+    PyArray_CLEARFLAGS(array, NPY_ARRAY_WRITEABLE);
+    if (PyArray_SetBaseObject(array, (PyObject *) self) != 0) {
+        goto out;
+    }
+    /* PyArray_SetBaseObject steals a reference, so we have to incref the tree
```
Comment (jeromekelleher) [inline python/_tskitmodule.c:11473]: out of date comment here, no tree

**Review by jeromekelleher — APPROVED** [review]
LGTM, spotted one small gap in testing.

File: python/_tskitmodule.c (hunk 2 above)
Comment (benjeffery) [reply python/_tskitmodule.c]: Now you know where I copy pasted from :smiley:! Fixed,

**Comment by benjeffery** [comment]
@jeromekelleher fixed up!

**Comment by jeromekelleher** [comment]
Looks good to go once linting CI is updated?

**Comment by benjeffery** [comment]
@mergifyio rebase

**Pull request merged**